lib/fastlane/actions/carthage.rb

```diff
@@ -68,20 +82,29 @@ def self.available_options
                                        verify_block: proc do |value|
                                          raise "Please pass a valid value for no_build. Use one of the following: true, false" unless value.kind_of?(TrueClass) || value.kind_of?(FalseClass)
                                        end),
-          FastlaneCore::ConfigItem.new(key: :verbose,
-                                       env_name: "FL_CARTHAGE_VERBOSE",
-                                       description: "Print xcodebuild output inline",
+          FastlaneCore::ConfigItem.new(key: :use_ssh,
+                                       env_name: "FL_CARTHAGE_USE_SSH",
+                                       description: "Use SSH for downloading GitHub repositories",
                                        is_string: false,
                                        optional: true,
                                        verify_block: proc do |value|
-                                         raise "Please pass a valid value for verbose. Use one of the following: true, false" unless value.kind_of?(TrueClass) || value.kind_of?(FalseClass)
+                                         raise "Please pass a valid value for use_ssh. Use one of the following: true, false" unless value.kind_of?(TrueClass) || value.kind_of?(FalseClass)
                                        end),
-          FastlaneCore::ConfigItem.new(key: :platform,
-                                       env_name: "FL_CARTHAGE_PLATFORM",
-                                       description: "Define which platform to build for",
+          FastlaneCore::ConfigItem.new(key: :use_submodules,
+                                       env_name: "FL_CARTHAGE_USE_SUBMODULES",
+                                       description: "Add dependencies as Git submodules",
+                                       is_string: false,
                                        optional: true,
                                        verify_block: proc do |value|
-                                         raise "Please pass a valid platform. Use one of the following: all, iOS, Mac, watchOS" unless ["all", "iOS", "Mac", "watchOS"].include? value
+                                         raise "Please pass a valid value for use_submodules. Use one of the following: true, false" unless value.kind_of?(TrueClass) || value.kind_of?(FalseClass)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :no_use_binaries,
+                                       env_name: "FL_CARTHAGE_NO_USE_BINARIES",
+                                       description: "Check out dependency repositories even when prebuilt frameworks exist",
+                                       is_string: false,
+                                       optional: true,
+                                       verify_block: proc do |value|
```

Comment:
What do you think about splitting the changes into separate PRs so that they can be discussed separately?
Personally I'm concerned about the breaking changes. I think it's great to support all the Carthage commands, but the rest of changes seem less important at this point, and it would be beat do discuss in a separate PR to go over a possible depreciation path.